### PR TITLE
Get it working

### PR DESCRIPTION
**Pull request by lewismarshall**
Tested manually on metal cluster.

Could squash but more for feedback on the whole project...

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,14 @@
-FROM alpine:3.8
+FROM golang:1.12
+WORKDIR /go/src/github.com/appvia/metal-pod-reaper/
+COPY . .
```
Comment (KashifSaadat) [inline Dockerfile]: Would require some restructuring of the repo, or a few `COPY` lines as you wouldn't want to add in everything by default (including readme, dot files, circleci etc.

Comment (KashifSaadat) [reply Dockerfile]: Actually not a big deal as you're using multi build process so it's not in the final image :)

File: pkg/detector/detector.go [hunk 2]
```diff
@@ -108,42 +109,61 @@ func (d *Detector) Run() error {
...
+				klog.V(4).Infof("about to check node %s with ip %s", nodeName, result.NetNode.IP)
 				nodeDown, err := isNodeDown(result.NetNode.IP)
 				// record the results
 				result.Err = err
 				result.IsNodeDown = nodeDown
+				if result.Err != nil {
+					klog.V(4).Infof("error checking node %s, %s", result.NetNode.IP, result.Err)
+				} else {
+					if nodeDown {
+						klog.V(4).Infof("node %s is unreachable, repeat NOT reachable", result.NetNode.IP)
```
Comment (KashifSaadat) [inline pkg/detector/detector.go]: Nitpick, this log line and the one above should probably be warning level or higher?

File: pkg/detector/detector.go [hunk 3]
```diff
@@ -108,42 +109,61 @@ func (d *Detector) Run() error {
...
+			klog.V(4).Infof("waiting for node result %d of %d", nodeIndex, len(checkableNodes))
 			nodeResult := <-results
+			klog.V(4).Infof("got node result %d of %d", nodeIndex, len(checkableNodes))
 			if nodeResult.Err != nil {
-				klog.Errorf("problem reporting on node ip %s: %s", node.NetNode.IP, nodeResult.Err)
+				klog.Errorf("problem reporting on node ip %s: %s", nodeResult.NetNode.IP, nodeResult.Err)
 			} else {
 				if nodeResult.IsNodeDown {
-					klog.V(1).Infof("unreachable node detetcted %s", node.NetNode.Node.Name)
+					klog.V(1).Infof("unreachable node detetcted %s", nodeResult.NetNode.IP)
```
Comment (KashifSaadat) [inline pkg/detector/detector.go]: `detetcted` spelling, probably log as a warning or error

**Comment by KashifSaadat** [comment]
2 small comments, all else looks good.

File: pkg/detector/detector.go (hunk 3 above)
Comment (lewismarshall) [reply pkg/detector/detector.go]: Fixed typo - made a warning 👍

File: pkg/detector/detector.go (hunk 2 above)
Comment (lewismarshall) [reply pkg/detector/detector.go]: Line 121 is normal operation, just debugging as the node down is reported at level v1 later. - the one above I've made  an error though 👍

File: pkg/detector/detector.go [hunk 4]
```diff
@@ -108,42 +109,61 @@ func (d *Detector) Run() error {
...
+					}
+				}
 				// Put the result on the channel (signal that the result is in)...
 				results <- result
 			}(node.NetNode.Node.Name)
 		}
 		var unReachableNodes []kubeutils.NetNode
 		// Now wait till the results are in for all nodes:
-		for _, node := range checkableNodes {
+		for nodeIndex := 1; nodeIndex <= len(checkableNodes); nodeIndex++ {
```
Comment (gambol99) [inline pkg/detector/detector.go]: `for i, _ := range checkableNodes {`

Comment (lewismarshall) [reply pkg/detector/detector.go]: ? https://play.golang.org/p/OePa5U1hwp7

**Pull request merged**